### ISSUE TITLE
Upgrading from 2.1.3 to 2.20 some dags fail due to pickeling

### ISSUE THREAD

**Issue by Jaxing** [apache/airflow#19040]
### Apache Airflow version

2.2.0 (latest released)

### Operating System

Kubernetes

### Deployment

Other

### Deployment details

Deploying using the docker image and a custom deployment file

### What happened

Most dags opens fine but some fails and shows this error page
```
Python version: 3.6.15
Airflow version: 2.2.0
Node: airflow-86c589ddb9-q9sdg
-------------------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.6/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/airflow/.local/lib/python3.6/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/airflow/.local/lib/python3.6/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/airflow/.local/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/airflow/.local/lib/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/airflow/.local/lib/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/www/auth.py", line 51, in decorated
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/www/decorators.py", line 109, in view_func
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/www/decorators.py", line 72, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/utils/session.py", line 70, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/www/views.py", line 2283, in tree
    .limit(num_runs)
  File "/home/airflow/.local/lib/python3.6/site-packages/sqlalchemy/orm/query.py", line 3373, in all
    return list(self)
  File "/home/airflow/.local/lib/python3.6/site-packages/sqlalchemy/orm/loading.py", line 100, in instances
    cursor.close()
  File "/home/airflow/.local/lib/python3.6/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    with_traceback=exc_tb,
  File "/home/airflow/.local/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
  File "/home/airflow/.local/lib/python3.6/site-packages/sqlalchemy/orm/loading.py", line 80, in instances
    rows = [proc(row) for row in fetch]
  File "/home/airflow/.local/lib/python3.6/site-packages/sqlalchemy/orm/loading.py", line 80, in <listcomp>
    rows = [proc(row) for row in fetch]
  File "/home/airflow/.local/lib/python3.6/site-packages/sqlalchemy/orm/loading.py", line 588, in _instance
    populators,
  File "/home/airflow/.local/lib/python3.6/site-packages/sqlalchemy/orm/loading.py", line 725, in _populate_full
    dict_[key] = getter(row)
  File "/home/airflow/.local/lib/python3.6/site-packages/sqlalchemy/sql/sqltypes.py", line 1723, in process
    return loads(value)
ValueError: unsupported pickle protocol: 5
```
No python versions has changed other than possibly the ones in the airflow image.

### What you expected to happen

Display the dag as usual 

### How to reproduce

Set up airflow 2.1.3 in kubernetes (will probably work with the helm charts). Create some dags and run some of them , not entirely sure what causes it but the one that fails has a default config that you can run it with. Then change airflow version to 2.2.0 and run the db upgrade and try to open tree view for you dags.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by Jaxing** [comment]
We have never been have never been explicitly on 3.6 that is if the default changed from 3.6 to 3.8?

**Comment by uranusjr** [comment]
> that is if the default changed from 3.6 to 3.8?

It did not, but even if it did there shouldn't be a problem. What seems to have happened is the other way around: The _2.1_ installation was on 3.8, but now the _2.2_ installation is using 3.6, which is the weird thing.

**Comment by Jaxing** [comment]
Hmm ok. I'm trying to reproduce it now, but I had to wipe our staging env when I downgraded from the broken 2.2 deployment so now when I upgrade it I cannot reproduce.

We are only using kubernetes operator and bigquery operators. And the airflow image versions have always been just the version number as far as I can see in our git history.

I'm going to down grade to 2.1.3 again try to run the dag that had trouble before and then upgrade again, if that's still ok than I cannot reproduce and we can close this issue for now?
